src/main/java/de/tum/in/www1/artemis/web/rest/metis/conversation/ConversationResource.java

```diff
@@ -107,6 +107,22 @@ public ResponseEntity<Void> switchHiddenStatus(@PathVariable Long courseId, @Pat
         return ResponseEntity.ok().build();
     }
 
+    /**
+     * GET /api/courses/:courseId/unread-messages : Checks for unread messages of the current user
+     *
+     * @param courseId the id of the course
+     * @return ResponseEntity with status 200 (Ok) and the information if the user has unread messages
+     */
+    @GetMapping("/{courseId}/unread-messages")
+    @PreAuthorize("hasRole('USER')")
+    public ResponseEntity<Boolean> hasUnreadMessages(@PathVariable Long courseId) {
+        checkMessagingEnabledElseThrow(courseId);
+
+        var requestingUser = userRepository.getUserWithGroupsAndAuthorities();
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.STUDENT, courseRepository.findByIdElseThrow(courseId), requestingUser);
```

Comment:
Auth mismatch. You expect at least a student but the PreAuthorization expects at least a user